core/server/xendorser/plugin-default/xendorser_impl_test.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/xuperchain/xuperchain/core/pb"
+)
+
+var baseDir = os.Getenv("XCHAIN_ROOT")
+
+type MorkXEndorserServer struct {
+	Txs map[string]*pb.TxStatus
+}
+
+func (s *MorkXEndorserServer) PostTx(context.Context, *pb.TxStatus) (*pb.CommonReply, error) {
+	return nil, nil
+}
+func (s *MorkXEndorserServer) QueryTx(context context.Context, txStatus *pb.TxStatus) (*pb.TxStatus, error) {
+	return s.Txs[string(txStatus.Txid)], nil
+}
+func (s *MorkXEndorserServer) PreExecWithSelectUTXO(context.Context, *pb.PreExecWithSelectUTXORequest) (*pb.PreExecWithSelectUTXOResponse, error) {
+	return nil, nil
+}
+func (s *MorkXEndorserServer) PreExec(context.Context, *pb.InvokeRPCRequest) (*pb.InvokeRPCResponse, error) {
+	return nil, nil
+}
+
+func TestGetTx(t *testing.T) {
+	os.Chdir(baseDir)
```

Comment:
Check `basedir` variable, if it is empty, skip the test

Reply:
done